File: python/0014.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        ml=min(map(lambda x:len(x),strs))
        ans=-1
        for i in range(ml):
            s=set()
            for j in strs:
                s.add(j[i])
            if len(s)!=1:
                ans=i
                break
        if ans==-1:ans=ml
        return strs[0][:ans]

File: rust/src/bin/0014.rs
use std::str::Chars;

struct Solution;


impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.len() == 0 {
            return "".to_string();
        }
        let mut s = "".to_string();
        let mut a = Vec::<Vec<char>>::new();
        for i in strs {
            a.push(i.chars().collect());
        }
        let mut j = 0;
        'out:
        loop {
            let mut cur: Option<char> = None;
            for i in &a {
                if j >= i.len() {
                    break 'out;
                }
                let now = i[j];
                if cur.is_some() && now != cur.unwrap() {
                    break 'out;
                } else {
                    cur = Some(now);
                }
            }
            s.push(cur.unwrap());
            j += 1;
        }
        s
    }
}

fn gets(a: Vec<&str>) -> Vec<String> {
    let mut b = Vec::<String>::new();
    for i in a {
        b.push(i.to_string());
    }
    b
}

fn main() {
    println!("{}", Solution::longest_common_prefix(gets(vec!["flower", "flow", "flight"])));
}